File: tkpdsession/src/main/java/com/tokopedia/profilecompletion/data/mapper/EditUserInfoMapper.java
package com.tokopedia.profilecompletion.data.mapper;

import com.tokopedia.core.network.retrofit.response.TkpdResponse;
import com.tokopedia.profilecompletion.data.pojo.EditUserInfoData;
import com.tokopedia.profilecompletion.domain.model.EditUserInfoDomainModel;

import retrofit2.Response;
import rx.functions.Func1;

/**
 * @author by nisie on 7/3/17.
 */

public class EditUserInfoMapper implements Func1<Response<TkpdResponse>, EditUserInfoDomainModel> {

    @Override
    public EditUserInfoDomainModel call(Response<TkpdResponse> response) {
        return mappingResponse(response);
    }

    private EditUserInfoDomainModel mappingResponse(Response<TkpdResponse> response) {
        EditUserInfoDomainModel model = new EditUserInfoDomainModel();
        if (response.isSuccessful()) {
            if (!response.body().isError()) {
                EditUserInfoData data = response.body().convertDataObj(EditUserInfoData.class);
                model.setSuccess(data.getIsSuccess() == 1);
            } else {
                if (response.body().getErrorMessages() == null
                        && response.body().getErrorMessages().isEmpty()) {
                    model.setSuccess(false);
                } else {
                    model.setSuccess(false);
                    model.setErrorMessage(response.body().getErrorMessageJoined());
                }
            }
            model.setStatusMessage(response.body().getStatusMessageJoined());
        } else {
            model.setSuccess(false);
        }
        model.setResponseCode(response.code());
        return model;
    }
}

File: tkpdsession/src/main/java/com/tokopedia/profilecompletion/data/mapper/EditUserInfoMapper.kt
package com.tokopedia.profilecompletion.data.mapper

import com.tokopedia.core.network.retrofit.response.TkpdResponse
import com.tokopedia.profilecompletion.data.pojo.EditUserInfoData
import com.tokopedia.profilecompletion.domain.model.EditUserInfoDomainModel
import retrofit2.Response
import rx.functions.Func1
import javax.inject.Inject

/**
 * @author by nisie on 7/3/17.
 */
class EditUserInfoMapper @Inject constructor() : Func1<Response<TkpdResponse>, EditUserInfoDomainModel> {
    override fun call(response: Response<TkpdResponse>): EditUserInfoDomainModel {
        return mappingResponse(response)
    }

    private fun mappingResponse(response: Response<TkpdResponse>): EditUserInfoDomainModel {
        val model = EditUserInfoDomainModel()
        if (response.isSuccessful) {
            if (!response.body()!!.isError) {
                val data = response.body()!!.convertDataObj(EditUserInfoData::class.java)
                model.isSuccess = data.isSuccess == 1
            } else {
                if (response.body()!!.errorMessages == null
                        && response.body()!!.errorMessages.isEmpty()) {
                    model.isSuccess = false
                } else {
                    model.isSuccess = false
                    model.errorMessage = response.body()!!.errorMessageJoined
                }
            }
            model.statusMessage = response.body()!!.statusMessageJoined
        } else {
            model.isSuccess = false
        }
        model.responseCode = response.code()
        return model
    }
}